Packs/ExportIndicators/Integrations/ExportIndicators/ExportIndicators.py

```diff
@@ -108,13 +119,90 @@ def find_indicators_with_limit_loop(indicator_query: str, limit: int, total_fetc
     return iocs, next_page
 
 
-def create_values_out_dict(iocs: list, out_format: str) -> dict:
+def panos_url_formatting(iocs: list, drop_invalids: bool, strip_port: bool):
+    formatted_indicators = []
+    for indicator_data in iocs:
+        # only format URLs and Domains
+        if indicator_data.get('indicator_type') in ['URL', 'Domain']:
+            indicator = indicator_data.get('value').lower()
+            # remove initial protocol
+            indicator = _PROTOCOL_RE.sub('', indicator)
+            indicator_with_port = indicator
+            # remove port from indicator
+            indicator = _PORT_RE.sub(r'\g<1>', indicator)
+
+            if indicator != indicator_with_port and not strip_port:
+                # if port was in the indicator and strip_port param not set - ignore the indicator
+                continue
+
+            with_invalid_tokens_indicator = indicator
+            # remove invalid tokens from indicator
+            indicator = _INVALID_TOKEN_RE.sub('*', indicator)
+            if with_invalid_tokens_indicator != indicator:
+                # invalid tokens in indicator- if drop_invalids is set - ignore the indicator
+                if drop_invalids:
+                    continue
+
+                # check if after removing the tokens the indicator is too broad if so - ignore
+                hostname = indicator
+                if '/' in hostname:
+                    hostname, _ = hostname.split('/', 1)
+
+                if _BROAD_PATTERN.match(hostname) is not None:
+                    continue
+
+            # for PAN-OS "*.domain.com" does not match "domain.com"
+            # we should provide both
+            # this could generate more than num entries in the egress feed
+            if indicator.startswith('*.'):
+                formatted_indicators.append(indicator[2:])
```

Comment:
This is a bit confusing can we make it more readable?

Reply:
am trying - it is a work in progress